system/jlib/jtrace.cpp

```diff
@@ -69,3 +100,688 @@ const char* LogTrace::queryLocalId() const
+            out.append("Server");
+        else if (type == SpanType::ClientSpan)
+            out.append("Client");
+        else
+            out.append("Internal");
+
+        if (span != nullptr)
+        {
+            out.append(" Name: ").append(name.get())
+            .append(" SpanID: ").append(spanID.get())
+            .append(" TraceID: ").append(traceID.get())
+            .append(" TraceFlags: ").append(traceFlags.get())
+            .append(" HPCCGlobalID: ").append(hpccGlobalId.get())
+            .append(" HPCCCallerID: ").append(hpccCallerId.get());
+
+            if (localParentSpan != nullptr)
+            {
+                out.append(" localParentSpan: ");
+                localParentSpan->toString(out);
+            }
+
+            if (type == SpanType::ServerSpan && remoteParentSpanCtx.IsValid())
+            {
+                out.append(" remoteParentSpanID: ");
+                char spanId[16] = {0};
+                remoteParentSpanCtx.span_id().ToLowerBase16(spanId);
+                out.append(16, spanId);
+            }
+        }
+    };
+
+    void setSpanAttributes(const IProperties * attributes) override
+    {
+        Owned<IPropertyIterator> iter = attributes->getIterator();
+        ForEach(*iter)
+        {
+            const char * key = iter->getPropKey();
+            if (!isEmptyString(key))
+                setSpanAttribute(key, attributes->queryProp(key));
+        }
+    }
+
+    void setSpanAttribute(const char * key, const char * val) override
+    {
+        if (span && !isEmptyString(key) && !isEmptyString(val))
+            span->SetAttribute(key, val);
+    }
+
+    void addSpanEvent(const char * eventName) override
+    {
+        if (span && !isEmptyString(eventName))
+            span->AddEvent(eventName);
+    }
+
+    bool getSpanContext(CHPCCHttpTextMapCarrier * carrier) const override
+    {
+        if (!carrier)
+            return false;
+
+        auto propagator = opentelemetry::context::propagation::GlobalTextMapPropagator::GetGlobalPropagator();
+
+        opentelemetry::context::Context empty;
+        auto spanCtx = SetSpan(empty, span); //Rodrigo, not sure if this is valid, need to test
+
+        //and have the propagator inject the ctx into carrier
+        propagator->Inject(*carrier, spanCtx);
+
+        if (!isEmptyString(hpccGlobalId.get()))
+            carrier->Set(HPCCSemanticConventions::kGLOBALIDHTTPHeader, hpccGlobalId.get());
+
+        if (!isEmptyString(hpccCallerId.get()))
+            carrier->Set(HPCCSemanticConventions::kCallerIdHTTPHeader, hpccCallerId.get());
+
+        return true;
+    }
+
+    /**
+     * Retrieves the OpenTelemetry formatted span context from the given properties object.
+     *
+     * @param contextProps The properties object containing the span context.
+     * @return True if the span context was successfully retrieved, false otherwise.
+     */
+    bool getSpanContext(IProperties * ctxProps, bool otelFormatted) const override
```

Comment:
client spans would set otelformatted == true in order to pass along otel context.
We could change behavior based on span type, but it feels cleaner to let developer declare the desired format